File: php/testcode/codeinj_021.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_codeinj_cookie_callback
function codeinj021(BenchmarkRequest $req): BenchmarkResponse {
    $handler = $req->cookie('handler');
    $data = $req->param('data');
    $result = call_user_func($handler, $data); // vuln-code-snippet vuln-line php_codeinj_cookie_callback
    return BenchmarkResponse::ok((string) $result);
}
// vuln-code-snippet end php_codeinj_cookie_callback

File: ruby/testcode/codeinj_021.rb
require_relative 'shared'

begin
  require 'erb'
rescue LoadError
end

# vuln-code-snippet start ruby_codeinj_erb_new_result
def render_template(req)
  tpl = req.param('tpl')
  renderer = ERB.new(tpl)
  output = renderer.result(binding) # vuln-code-snippet vuln-line ruby_codeinj_erb_new_result
  BenchmarkResponse.html(output)
end
# vuln-code-snippet end ruby_codeinj_erb_new_result
